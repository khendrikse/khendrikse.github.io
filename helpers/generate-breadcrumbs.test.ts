import generateBreadcrumbs from './generate-breadcrumbs';

describe('generateBreadcrumbs', () => {
  it('generates breadcrumbs structured data', () => {
    const input = [{ name: 'a name', item: 'thing' }];
    expect(generateBreadcrumbs(input)).toBe(
      // eslint-disable-next-line max-len
      '{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"home","item":"https://khendrikse.github.io/"},{"@type":"ListItem","position":2,"name":"a name","item":"https://khendrikse.github.io/thing"}]}'
    );
  });

  it('generates breadcrumbs structured data with only home', () => {
    expect(generateBreadcrumbs()).toBe(
      // eslint-disable-next-line max-len
      '{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"home","item":"https://khendrikse.github.io/"}]}'
    );
  });
});
