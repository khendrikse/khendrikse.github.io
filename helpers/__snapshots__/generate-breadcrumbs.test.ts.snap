// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateBreadcrumbs generates breadcrumbs structured data 1`] = `"{\\"@context\\":\\"https://schema.org/\\",\\"@type\\":\\"BreadcrumbList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"name\\":\\"home\\",\\"item\\":\\"https://khendrikse.github.io/\\"},{\\"@type\\":\\"ListItem\\",\\"position\\":2,\\"name\\":\\"a name\\",\\"item\\":\\"https://khendrikse.github.io/thing\\"}]}"`;

exports[`generateBreadcrumbs generates breadcrumbs structured data with only home 1`] = `"{\\"@context\\":\\"https://schema.org/\\",\\"@type\\":\\"BreadcrumbList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"name\\":\\"home\\",\\"item\\":\\"https://khendrikse.github.io/\\"}]}"`;
